tests/python/gaia-ui-tests/gaiatest/tests/functional/lockscreen/test_lockscreen_time_check.py

```diff
@@ -0,0 +1,45 @@
+import time
+
+from marionette.wait import Wait
+
+from gaiatest import GaiaTestCase
+from gaiatest.apps.lockscreen.app import LockScreen
+
+
+class TestLockScreen(GaiaTestCase):
+
+    _seconds_since_epoch = 1357043430  # the time is 12:30 GMT
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        #self.connect_to_local_area_network()
+
+    def test_lockscreen_time_check(self):
+
+        # disable time/timezone automatic update
+        self.data_layer.set_setting('time.clock.automatic-update.enabled', False)
+        self.data_layer.set_setting('time.timezone.automatic-update.enabled', False)
+
+        # set time manually to 12:30
+        self.data_layer.set_time(self._seconds_since_epoch * 1000)
+        self.data_layer.set_setting('time.timezone', 'Atlantic/Reykjavik')
+
+        self.device.lock()
+        lock_screen = LockScreen(self.marionette)
+
+        #check the time on the lockscreen and unlock.  Below check gives 10 minutes of windows
+        #on desktop b2g, it usually takes 8 minutes to get to this point
+        self.assertTrue('12:3' in lock_screen.time)
```

Comment:
hmm, I should try that, this might take a bit of a time, but i guess that makes it more faithfully reproducing the steps in the bug.  I am not sure how the try server tests work, because by the time it reaches line 32, lockscreen time was showing 12:38.